hubploy/auth.py

```diff
@@ -188,7 +217,11 @@ def cluster_auth(deployment):
                 raise ValueError(
                     f'Unknown provider {provider} found in hubploy.yaml')
         finally:
-            unset_env_var("KUBECONFIG", orig_kubeconfig)
+            # unset KUBECONFIG if it is set
+            try:
+                unset_env_var("KUBECONFIG", orig_kubeconfig)
```

Comment:
Can you tell me what exception was being raised here?

Reply:
Unfortunately, I don't remember :(

I think it was `del os.environ[env_var]` in the `unset_env_var` function was failing.  I don't know if I should check for it's existence in the environment, or if there is the expectation that a particular variable always exists.

It is possible that it will be set to None in `cluster_auth`.

Reply:
Hmm, ok. I'd suggest we remove this, and see if the error comes back? Silently `pass`ing exceptions usually causes problems later.